api/apps/api/test/files-repository/files-repository.e2e-spec.ts

```diff
@@ -0,0 +1,75 @@
+import { createReadStream } from 'fs';
+import { Readable } from 'stream';
+import { isRight, Right } from 'fp-ts/Either';
+import { Test } from '@nestjs/testing';
+
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import {
+  FileRepository,
+  FileRepositoryModule,
+  TempStorageRepository,
+} from '@marxan/files-repository';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+
+afterEach(async () => {
+  await fixtures?.cleanup();
+});
+
+/**
+ * Please note that bootstrapApplication overrides default implementation
+ * with `/tmp/storage/` usage
+ */
+test(`adding a file`, async () => {
+  const uri = await fixtures.GivenFileWasAdded();
+  const file = await fixtures.WhenGettingFile(uri);
+  await fixtures.ThenFileIsRetrieved(file);
+});
+
+const getFixtures = async () => {
+  /**
+   * overriding repository implementation happens while bootstrapping test app
+   * to avoid communicating with potential storage provider
+   */
+  const app = await Test.createTestingModule({
+    imports: [FileRepositoryModule],
+  })
+    .overrideProvider(FileRepository)
+    .useClass(TempStorageRepository)
+    .compile();
+  const sut = app.get(FileRepository);
+
+  return {
+    cleanup: async () => app.close(),
```

Comment:
Not removing the files - see PR description. Shouldn't be a big deal for development and CI.